packages/dds/map/src/map.ts

```diff
@@ -136,14 +141,24 @@ export class SharedMap extends SharedObject<ISharedMapEvents> implements IShared
 		id: string,
 		runtime: IFluidDataStoreRuntime,
 		attributes: IChannelAttributes,
+		options?: IMapOptions,
 	) {
 		super(id, runtime, attributes, "fluid_map_");
+		this.options ??= options;
+
+		const configSetAttribution = loggerToMonitoringContext(this.logger).config.getBoolean(
+			"Fluid.Attribution.EnableOnNewFile",
+		);
+		if (configSetAttribution !== undefined) {
+			(this.options ?? (this.options = {})).attribution = { track: configSetAttribution };
```

Comment:
this could override other attribution-related options in the loader. The whole pattern here is ugly and I was meaning to get to a cleanup item which I realized is pretty easy. Once #14006 is in, you should just be able to pass the options directly (and ignore config) here.